mkat_fpga_tests/test_cbf.py

```diff
@@ -1756,3 +1774,217 @@ def get_data_switch():
         Aqf.step('CBF ROACH information on each Data Switch.')
         get_data_switch()
         Aqf.hop('Test ran by: {} on {}'.format(os.getlogin(), time.ctime()))
+
+    @aqf_vr('TP.C.1.18')
+    def test_fault_detection(self):
+        """AR1 Fault detection"""
+
+        def air_temp_warn(io_dir, label):
+
+            hwmon_dir = '/sys/class/hwmon/hwmon{}'.format(io_dir)
+            # returns current temperature
+            read_cur_temp = 'cat {}/temp1_input\n'.format(hwmon_dir)
+
+            # returns 1 if the roach is overtemp, it should be 0
+            read_overtemp_ind = 'cat {}/temp1_max_alarm\n'.format(hwmon_dir)
+            # returns 0 if the roach is undertemp, it should be 1
+            read_undertemp_ind = 'cat {}/temp1_min_alarm\n'.format(hwmon_dir)
+
+            # set the max temp limit to 10 degrees
+            set_max_limit = 'echo "10000" > {}/temp1_max\n'.format(hwmon_dir)
+            # set the min temp limit to below current temp
+            set_min_limit = 'echo "10000" > {}/temp1_min\n'.format(hwmon_dir)
```

Comment:
The comment seems to be wrong since you are not using the current temp anywhere? I don't quite understand the logic of the temperatures values you are setting, could you perhaps help me in person?